examples/todo/src/datasources/db.datasource.ts

```diff
@@ -7,13 +7,56 @@ import {inject} from '@loopback/core';
 import {juggler, DataSource} from '@loopback/repository';
 const config = require('./db.datasource.json');
 
+type VcapService = {
+  name: string;
+  credentials: {
+    // tslint:disable-next-line:no-any
+    [property: string]: any;
+  };
+};
+
+if (shouldConfigureForIbmCloud()) {
+  const services = JSON.parse(process.env.VCAP_SERVICES!);
+  Object.keys(services).forEach(key => {
+    const serviceGroup = services[key];
+
+    // tslint:disable-next-line:no-any
+    serviceGroup.forEach((service: VcapService) => {
+      if (service.name === config.name) {
```

Comment:
Have you considered using an npm library to handle VCAP_SERVICES processing for you?

Few examples found by my quick search:
 - https://www.npmjs.com/package/vcap_services
 - https://www.npmjs.com/package/cf-services

The latter package says in its README:  _Unfortunately the instance name is rarely known in advance, so you may pass it as a separate environment variable._ Is this something we need to take care of?

Reply:
I haven't looked at them, since I was focussed on just making the app work. Will take a look at them.